go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/googlecloudplatform/gcsfuse/v2
 
-go 1.23.0
+go 1.23.2
```

Comment:
Why do we need to change this? The source should still be compatible with golang 1.23.0, right?

Reply:
This ensures the `package_gcsfuse` Dockerfile/end user/developer always compiles GCSFuse with Go 1.23.2, which includes important security fixes. Since end users are not expected to compile GCSFuse themselves, this restriction should not cause any issues IMO. WDYT?

Reply:
I think it's not optimal to update the source version and making users download/install a newer version of GoLang everytime. Development and release environments are also not guaranteed to be exactly the same even if we upgrade the go.mod version since if the development environment has a more recent version of Golang installed, that'll be used instead of what's specified in the go.mod.
TBH, it's not a big deal for us since GCSFuse is a binary and not a library, so I'm okay approving it.